### DISCUSSION TITLE
Fooocus 2.1.60: Official Anime and Realistic Support

### DISCUSSION THREAD

**Discussion by lllyasviel** [lllyasviel/Fooocus#679] [Announcements]
After 2.1.60, Fooocus begin to officially support Anime and Realistic models. (Previously this is achieved by users but relatively difficult to get best results, now they only require one click.)

Now we will have 3 launchers:
![image](https://github.com/lllyasviel/Fooocus/assets/19834515/577a03bc-53d3-4f15-a160-e48996c9efbd)

If you still only have one launcher, you can run “run.bat” and it will automatically update and give you all three launchers.

# All below results are using seed 12345 and default settings

**Below are real performance. Seeds are not cherry-picked.**

![image](https://github.com/lllyasviel/Fooocus/assets/19834515/432f05e8-c47a-448b-9705-8a6c31ca6016)

**Below images share similarity (like same characters, same poses, etc) because they use same seeds. This is not a model problem.**

# Fooocus Anime

### 1girl, garden

![image](https://github.com/lllyasviel/Fooocus/assets/19834515/166434f0-1ff3-4a03-876f-cdaa32be3604)
![image](https://github.com/lllyasviel/Fooocus/assets/19834515/fad55906-7f5e-4183-9af8-66c8750e293f)
![image](https://github.com/lllyasviel/Fooocus/assets/19834515/2416761f-ae65-429e-8dd6-168bd2446028)

### 1girl, dragon

![image](https://github.com/lllyasviel/Fooocus/assets/19834515/30f4ad4b-ef09-496a-b052-5b1394d040b3)
![image](https://github.com/lllyasviel/Fooocus/assets/19834515/fbdb10a1-ff6a-4472-b8b6-596a5ed57b15)
![image](https://github.com/lllyasviel/Fooocus/assets/19834515/55d138ec-5b6b-4ac3-93f2-7e384582c486)

### 1girl, in classroom

![image](https://github.com/lllyasviel/Fooocus/assets/19834515/53c30315-3f4c-4522-8cb9-fbddbe0cd518)
![image](https://github.com/lllyasviel/Fooocus/assets/19834515/8b9be732-87e2-4e90-9927-c62e62c57903)
![image](https://github.com/lllyasviel/Fooocus/assets/19834515/ab9aafd4-a2f6-44db-b91b-61cbac9ff3b4)

### 1girl, dark magic

![image](https://github.com/lllyasviel/Fooocus/assets/19834515/422eb043-d27f-4d1d-8e80-8ab5dee45265)
![image](https://github.com/lllyasviel/Fooocus/assets/19834515/45923890-cee2-4281-859c-d4ca1279fa75)
![image](https://github.com/lllyasviel/Fooocus/assets/19834515/6a03331a-bc78-4119-8d6f-7047f8b0cd94)

### 1girl, princess in the castle

![image](https://github.com/lllyasviel/Fooocus/assets/19834515/c24ba67c-0baa-4eda-8db1-01f197c7a160)
![image](https://github.com/lllyasviel/Fooocus/assets/19834515/da37ac96-d7ff-4a39-9269-c6edd967a506)
![image](https://github.com/lllyasviel/Fooocus/assets/19834515/bda0cbb2-6b54-4efd-9a87-4c2ca2c053bb)

# Fooocus Realistic

<a name="realistic"></a>

### portrait of old man in the room, closeup

![image](https://github.com/lllyasviel/Fooocus/assets/19834515/e169e841-6759-46ef-9b66-c89cdd72aea9)
![image](https://github.com/lllyasviel/Fooocus/assets/19834515/95151fd1-659d-47ed-b362-78f046629409)
![image](https://github.com/lllyasviel/Fooocus/assets/19834515/46f81fdd-fe5b-4408-9a55-543ceff69787)

### a woman, New York city

![image](https://github.com/lllyasviel/Fooocus/assets/19834515/c0f8dc56-2bc6-4d78-a934-77ccdf3510e2)
![image](https://github.com/lllyasviel/Fooocus/assets/19834515/3838504e-9ec5-458a-ad7f-58faca29e725)
![image](https://github.com/lllyasviel/Fooocus/assets/19834515/96643ac1-d2cd-4b29-9bcd-560632716c1a)

### a girl with beautiful eyes, closeup

![image](https://github.com/lllyasviel/Fooocus/assets/19834515/de65e091-c6fa-41f2-8096-9478c8e24038)
![image](https://github.com/lllyasviel/Fooocus/assets/19834515/1e442c4d-85aa-4609-a73f-17321cf357e6)
![image](https://github.com/lllyasviel/Fooocus/assets/19834515/8fa17566-231d-43c6-bd10-a94896dd3894)

### The man in the forest, portrait

![image](https://github.com/lllyasviel/Fooocus/assets/19834515/3c9e1533-dd53-4236-b327-91ae1e67760e)

![image](https://github.com/lllyasviel/Fooocus/assets/19834515/584d76d5-8824-4790-8ac4-7ff5a6e67cb7)

![image](https://github.com/lllyasviel/Fooocus/assets/19834515/1564fca7-a2d6-4345-bfdc-bbca77408b8f)

# Linux or Pure Python

You can use thse cmd lines if you do not use bat:

    python entry_with_update.py --preset anime
    python entry_with_update.py --preset realistic

If you do not want the scripts to download models for you, the URLs of models are here:

Fooocus Anime use SD1.5 (DreamShaper_8) to refiner SDXL (bluePencilXL), note that the "sd1.5-as-xl-refiner algorithm" is different from other software - it is Fooocus-only. Below are model URLs:

```json
    "checkpoint_downloads": {
        "bluePencilXL_v050.safetensors": "https://huggingface.co/lllyasviel/fav_models/resolve/main/fav/bluePencilXL_v050.safetensors",
        "DreamShaper_8_pruned.safetensors": "https://huggingface.co/Lykon/DreamShaper/resolve/main/DreamShaper_8_pruned.safetensors"
    },
    "embeddings_downloads": {
        "unaestheticXLv31.safetensors": "https://huggingface.co/lllyasviel/fav_models/resolve/main/fav/unaestheticXLv31.safetensors"
    },
    "lora_downloads": {
        "sd_xl_offset_example-lora_1.0.safetensors": "https://huggingface.co/stabilityai/stable-diffusion-xl-base-1.0/resolve/main/sd_xl_offset_example-lora_1.0.safetensors"
    }
```

Fooocus Realistic use Fooocus sharpness to enhance SDXL (realisticStockPhoto): 

```json
    "checkpoint_downloads": {
        "realisticStockPhoto_v10.safetensors": "https://huggingface.co/lllyasviel/fav_models/resolve/main/fav/realisticStockPhoto_v10.safetensors"
    },
    "lora_downloads": {
        "SDXL_FILM_PHOTOGRAPHY_STYLE_BetaV0.4.safetensors": "https://huggingface.co/lllyasviel/fav_models/resolve/main/fav/SDXL_FILM_PHOTOGRAPHY_STYLE_BetaV0.4.safetensors"
    }
```

**Comment by 3Diva** [comment]
Wow! Two (or three?) versions of Fooocus!? That sounds quite ambitious! Can we use the regular "run.bat" if we enjoy creating both types of images in the same session and don't want to switch between different versions?

**Reply by xhoxye** [reply]
The results are similar to if you downloaded the model yourself and adjusted the prompt parameters, but the author has his own experience.
There are preset files under the folder, which are called at startup time via launcher parameters, --preset anime
Fooocus\presets
anime.json
realistic.json
sdxl.json

civitai
Fooocus Anime：
base XL：https://civitai.com/models/119012/bluepencil-xl
refiner 1.5：https://civitai.com/models/4384/dreamshaper
embeddings：https://civitai.com/models/119032/unaestheticxl-or-negative-ti

Fooocus Realistic：
base XL：https://civitai.com/models/139565/realistic-stock-photo
LORA：https://civitai.com/models/158945/sdxl-film-photography-style

**Comment by ghost** [comment]
how to make for gamedev?

**Comment by lllyasviel** [comment]
if you want to create your own preset, and you know what you are doing and really want to create your own preset, you can look at

https://github.com/lllyasviel/Fooocus/tree/main/presets

then modify or create any bat into `python entry_with_update.py --preset name_of_your_preset`

but we do not recommend most users to modify files. This is undocumented and only works when users know how.

**Reply by eric23** [reply]
Can only attributes included in the file be used?

**Comment by TINZ** [comment]
This is a wonderful tool and I am having so much fun ... thank you! Just a teeny ask ... at some point, could you add the resolution 1720, 720? I have a 3440 x 1440 monitor and I am highly addicted to generating fantasy wallpapers for my rig. I have been adding a line to sdxl_styles.py, but the system (correctly) updates the file after my edits to maintain integrity. Marvellous work, and let me know how I can support. :)

**Reply by lllyasviel** [reply]
sdxl is not trained on 720p
users need to manually crop/upscale/outpaint it
editing codes to add 720 may work but will make results worse because sdxl is not trained on 720p

**Reply by TINZ** [reply]
Thank you, I will stick to the proper parameters. :)

**Comment by xhoxye** [comment]
It would be better to be able to switch between anime and realistic presets in one click in the UI interface

**Reply by AkoZ** [reply]
and also with the not automatic opening browser (with full tabs in it  which down all the speed and may crash the browser too)

**Comment by hzeasy** [comment]
Thank you for your work in making such great drawing tools available to us! ! grateful！
@lllyasviel 
If I want to keep using version 2.1.60, how can I prevent RUN from automatically upgrading? It is very important for users with poor network，Allow users to selectively update！

**Reply by 3Diva** [reply]
Fooocus-MRE has a run-mre-no-update file that does just that and I'd love to see Fooocus get something similar.

**Reply by hzeasy** [reply]
Regarding the functional convenience of MRE, I have to say that its "load prompt" function is indeed very practical, which greatly saves time in each creation. I think fooocus should add this feature.
![image](https://github.com/lllyasviel/Fooocus/assets/136664755/7296b1a2-a938-4307-a392-c600bc8cda1b)

**Comment by eric23** [comment]
It seems what I want, owning my personal preset is better.

**Comment by lanyusan** [comment]
@lllyasviel 

This is brilliant. I suggest to draft a protocol for preset json and prepare a more thorough doc for preset and put it at the repo's front page. 

And also I suggest to add a preset list / preview / switching UI. Artists can provide preview samples. 

This would encourage artists to create presets for different purposes, for example a preset for logo, a preset for illustrations etc. They could prepare pull request for presets.

I can see a whole creative community around this.

**Comment by MontyFlange** [comment]
I  keep getting bluePencilXL_v050.safetensors corrupt error when downloading the anima version

**Comment by pengpengzi** [comment]
I downloaded and tested three different launch methods, and they all worked well. However, I'm not quite sure why the style control is done through the launch methods. Wouldn't it be simpler to switch styles within the page itself?

**Reply by hijarian** [reply]
The base models used are different, and there's quite a lot of tweaks to do. bat files basically just save your time on reconfiguring.

**Comment by zzzPitoNzzz** [comment]
Hello! Thank you for your work, everything is great! But, not without complaints. I dragged a bunch of LORА, some of them are generally under the interface (see the screenshot), and without the preview, everything got mixed up, I would like to have a preview for them. As with styles, so much just doesn't fit in your head, and you need to know what you're turning on. Once again, thank you very much! 
![косякслорами1](https://github.com/lllyasviel/Fooocus/assets/114881552/1f71078c-1a35-40be-9435-b4db34412f2f)

**Comment by a1z2e3z4** [comment]
![Screenshot 2023-12-18 144614d](https://github.com/lllyasviel/Fooocus/assets/132886430/6bc561e8-27fd-4136-b90e-030456ed2fdf)
why download stuck to me :( ?

**Reply by NO000006** [reply]
大模型下载，因为你使用的大模型缺失。

**Reply by 3Diva** [reply]
Yeah, I would try what @liuhao985 suggested and manually download the Juggernaut XL version 6 model from CivitAI. You may have internet connection issues keeping Fooocus from downloading it properly.

Make sure you put that JuggernautXL V6 model in your Fooocus folder under Models >> Checkpoints. :)

**Comment by Hangover3832** [comment]
Fooocus = Fun. Just endless fun.

**Comment by JAL-code** [comment]
Downloaded the latest update and now Fooocus is working on an AMD computer with 8GB card. Two months ago, the code crashes when the model is passed to the GPU. Must be the change that uses the AMD version of torch.  Don't have more than one window open at a time.  

Anime Surfer Girl
![image](https://github.com/lllyasviel/Fooocus/assets/70151426/f9104ef7-d1ba-465e-9c6e-15e54866e69c)

Dogs at the Beach
![dogs_beach_02](https://github.com/lllyasviel/Fooocus/assets/70151426/2f5345cc-0765-4a9b-bee3-add1e21fd294)

Hand on a table with the correct number of fingers.  Three attempts before the picture was correct.
![hand_table_01](https://github.com/lllyasviel/Fooocus/assets/70151426/48770980-bdb0-4387-b21b-7c1a2376ee4b)

**Comment by picstudio** [comment]
Шикарный проект, жаль многие модели из https://civitai.com/ не поддерживаются

**Reply by mashb1t** [reply]
*to be exact all 1.5 and 2.1 models, as Fooocus is a SDXL only tool!

**Comment by Tahoon00** [comment]
I downloaded and installed the program, but I cannot access it. It is not in the Windows menu or on the desktop. How do I open the program on the computer?

**Reply by mashb1t** [reply]
@Tahoon00 see https://github.com/lllyasviel/Fooocus?tab=readme-ov-file#windows

> After you download the file, please uncompress it and then run the "run.bat".

**Comment by MLavrentii** [comment]
Hi guys. 
Is it possible to install 3 versions of fooocus? I installed realistic thought run_realistic.bat. I checked, and it works on my laptop.
then I tried to install another version, but it was stuck in the same place when I tried to install it through run_anime.bat it was tucked on 62%, and when I tried run.bat it stacked on 60%. I tried to install it some time and it stopped every time in the same percentages.
Also, I saw other models in the web interface, will they work too?

**Reply by rozhl** [reply]
The install process is mostly automated. You choose what you want inside the app. Other .bat files are just different options. From the symptoms you described, you have low free space on your main C volume. The app uses it no matter where you place it. In my case it used over 50 GB on the first run on C + around 16 GB on the install volume. Also the install fails when your internet connection gets disrupted.

**Comment by Melindro666** [comment]
Hi, this works nicely
But I'm trying to make anime characters for a game with rpg maker, and there's no way to make guys with masculine features, i'm only getting waifu girls. And every character wears earrings even though i am writting "earrings" in the negative prompt. 
This is the closest thing i got to what i want, and still is way far from it. You can see in the guy how his chest is slightly bloated, from trying hard to not put huge breasts on it. Also, i tried to make it without background, but there's no way, what am i doing wrong? 
![image](https://github.com/user-attachments/assets/7d58eb6b-f9fd-47cf-ae13-b8eb0dbb0d96)

**Reply by Rogala** [reply]
[Try other models and LoRAa](https://civitai.com/)

**Comment by DG88471** [comment]
Das geht gar nicht die BAT Dateien zeigen jedes mal Error an, wäre besser wenn da eine Exe mit dabei wäre wo man es normal starten kann.
